packages/SwingSet/src/kernel/kernelSyscall.js

```diff
@@ -105,6 +105,12 @@ export function makeKernelSyscallHandler(tools) {
     return OKNULL;
   }
 
+  function dropImports(koids) {
```

Comment:
Doesn't this need to also take the vatID as a parameter, since it's removing an export from the invoking vat's c-list?  Obviously it doesn't matter in the case of this no-op placeholder implementation, but for the real version it seems necessary, and the placeholder API should match the intended API.

Reply:
There are two steps that have to happen: remove the `vref-kref` mappings from the calling vat's c-list, and triggering GC activity afterwards. I'm removing the mappings during the translation phase, in `translateDropImports` (while we still have the `vref`, and still know which vat is calling). So by the time we hit this kernel-side no-longer-any-particular-vat-side `dropImports`, all that's left is the GC activity. My plan is to have this `dropImports` call some not-yet-written "examine all of the following krefs to see if their refcount is zero, if so, drop and recurse" function. I expect it will be more efficient for that call to take the full set of known-changed krefs, so having `dropImports` process the whole bunch, instead of having `deleteCListEntry` do them one at a time, might be better.

Although that approach does sound slightly weird, now that I write it out.. I could be talked into a better one.

Reply:
After sleeping on this I think I've come back around to my original view: `dropImports` should take the vatID as a parameter.  In particular, upon reflection I think that doing the c-list entry deletions in the translate layer is a modularity issue.  It's strange to do the work of a syscall there.  While the translate layer does _add_ c-list entries, this is done as a normal consequence of translation itself -- sometimes you need to do an import or an export to make the translation possible.  But other manipulations ought to be done in the syscall itself, and indeed many of the other syscalls take the vatID as a parameter for that purpose.